clientAkkaHttp/src/main/scala/RPCClient.scala

```diff
@@ -0,0 +1,75 @@
+package wiro.client
+
+import wiro.models.Config
+import wiro.server.akkaHttp.{ MethodMetaData, MetaDataMacro, PathMacro, OperationType }
+
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+
+import akka.stream.ActorMaterializer
+import akka.actor.ActorSystem
+
+import scala.concurrent.{ Future, ExecutionContext }
+
+import io.circe._
+import io.circe.syntax._
+import de.heikoseeberger.akkahttpcirce.CirceSupport._
+
+trait RPCClientContext[T] extends MetaDataMacro with PathMacro {
+  def methodsMetaData: Map[String, MethodMetaData]
+  def tp: Seq[String]
+  def path: String = tp.last
+}
+
+class RPCClient(
+  config: Config,
+  ctx: RPCClientContext[_]
+)(implicit
+  system: ActorSystem,
+  materializer: ActorMaterializer,
+  executionContext: ExecutionContext
+) extends autowire.Client[Json, Decoder, Encoder] {
+  def write[Result: Encoder](r: Result): Json = r.asJson
+
+  def read[Result: Decoder](p: Json): Result = {
+    //This trick is required to match the Result type of autowire
```

Comment:
Perhaps it would be better to move this code together with its  server side counterpart.